lang/cs/Org.Apache.REEF.IO/TempFileCreation/ITempFileCreator.cs

```diff
@@ -0,0 +1,43 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.IO.TempFileCreation
+{
+    /// <summary>
+    /// Provide interface to wrap temp file folder and temp file folder creation
+    /// </summary>
+    [DefaultImplementation(typeof(TempFileCreator))]
+    public interface ITempFileCreator
+    {
+        /// <summary>
+        /// return the temp folder path
+        /// </summary>
+        string TempFileFolder { get; }
+
+        /// <summary>
+        /// CReate a temp folder with given prefix and suffix as part of the name
+        /// </summary>
+        /// <param name="prefix"></param>
+        /// <param name="suffix"></param>
+        /// <returns></returns>
+        string CreateTempDirectory(string prefix = "", string suffix = "");
```

Comment:
The instance variable in this object is temp file folder rather than temp file name. The temp file folder is what we allow the user to set in configuration. With the same TempFileCreator, the user can call CreateTempDirectory to create different subfolders. So far, this interface/class is for temp folder not for a particular file. May be I should change the name to TempFIleFolderCreator?